arviz/plots/kdeplot.py

```diff
@@ -227,7 +229,8 @@ def plot_kde(
             fill_func(fill_x, fill_y, **fill_kwargs)
 
         if legend and label:
-            ax.legend()
+            legend_element = [Patch(edgecolor=default_color, label=label)]
+            ax.legend(handles=legend_element)
```

Comment:
This only overrides the previous behavior to always show a square of color `default_color` in the legend. In my opinion, 2 different cases must be detected to generate the legend in different ways.

1. `alpha=0` in `fill_kwargs`: The legend should show a line, not a rectangle. This is what the code used to do in all the cases. It can be left like that or it can be done using a `Line2D` object.
1. `alpha!=0` in `fill_kwargs`: the legend should show a rectangle, whose `facecolor` should be the color in `fill_kwargs`, `edgecolor` is the color in `plot_kwargs`, and both `fill_kwargs` and `plot_kwargs` should be passed (at least partially) to the `Patch` object. This is the reason why I would recommend using `linewidth=0` instead of `alpha=0` to hide line plots (and alpha for area/fill plots).